src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -49,6 +52,55 @@ public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessag
         /// </summary>
         protected string Namespace { get; private set; }
 
+        /// <summary>
+        /// Gets the unique identifier for this background job to distinguish this job instance in a multi-instance deployment.
+        /// </summary>
+        public string JobId { get; }
+
+        /// <summary>
+        /// Gets the name of the topic subscription; combined from the <see cref="AzureServiceBusMessagePumpSettings.SubscriptionPrefix"/> and the <see cref="JobId"/>.
+        /// </summary>
+        protected string SubscriptionName { get; }
+
+        /// <summary>
+        /// Triggered when the application host is ready to start the service.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (Settings.Options.TopicSubscription.HasFlag(TopicSubscription.CreateOnStart))
+            {
+                await CreateTopicSubscriptionAsync(cancellationToken);
+            }
+
+            await base.StartAsync(cancellationToken);
+        }
+
+        private async Task CreateTopicSubscriptionAsync(CancellationToken cancellationToken)
+        {
+            ServiceBusConnectionStringBuilder serviceBusConnectionString = await GetServiceBusConnectionStringAsync();
+
+            Logger.LogTrace(
+                "[Job: {JobId}] Creating subscription '{SubscriptionName}' on topic '{TopicPath}'...",
+                JobId, SubscriptionName, serviceBusConnectionString.EntityPath);
+            
+            var subscriptionDescription = new SubscriptionDescription(serviceBusConnectionString.EntityPath, SubscriptionName)
+            {
+                UserMetadata = $"Subscription created by Arcus job: '{JobId}' to process Service Bus messages."
+            };
+
+            var ruleDescription = new RuleDescription("Accept-All", new TrueFilter());
+            var serviceBusClient = new ManagementClient(serviceBusConnectionString);
+            await serviceBusClient.CreateSubscriptionAsync(subscriptionDescription, ruleDescription, cancellationToken)
```

Comment:
YOu shuld check if the subscription with same name already exists

Reply:
Yep done. Thanks.